### PR TITLE
OD-356 [Fix] Users will be able to work with accessibility

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://weboo.atlassian.net/browse/OD-356

## Description
Fixed accessibility issues from the table.

## Screenshots/screencasts
https://share.getcloudapp.com/RBu4BlWG

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko @ivandevupp @AndrRyaz

### CODE REVIEW

File: css/menu.css [hunk 1]
```diff
@@ -415,3 +415,7 @@ html.no-history .fl-viewport-header .nav-left {
 .has-overlay-menu {
   overflow-y: hidden;
 }
+
+.hidden {
```
Comment (tonytlwu) [inline css/menu.css:419]: Why is this needed?

image

Bootstrap already has this exact same style.

File: build.html [hunk 2]
```diff
@@ -41,16 +41,9 @@
 <!-- END OF HEADER -->
 <!-- MENU OVERLAY -->
 <div
-  class="fl-menu fl-app-menu {{#if location}}fl-menu-default-left{{else}}fl-menu-default{{/if}} {{#if location}}{{#if canGoBack}}hasGoBack{{/if}}{{/if}}">
+  class="fl-menu fl-app-menu {{#if location}}fl-menu-default-left{{else}}fl-menu-default{{/if}} {{#if location}}{{#if canGoBack}}hasGoBack{{/if}}{{/if}} hidden">
```
Comment (tonytlwu) [inline build.html:44]: https://share.getcloudapp.com/Jru1ye95

See above how the menu transitions in when the menu appears. It fades in and moves downwards slightly. With `hidden` applied and effectively adds `display: none`, I fear that the transition would be lost. Can you confirm that the transition still works?

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: build.html (hunk 2 above)
Comment (YaroslavOvdii) [reply build.html:44]: @tonytlwu Yes, you are right the transition is lost now. I'll add a new commit to fix that.

File: css/menu.css (hunk 1 above)
Comment (YaroslavOvdii) [reply css/menu.css:419]: @tonytlwu When I tried simply add a hidden class it has no effect. Should Bootrap do be added as a dependency?

Comment (tonytlwu) [reply css/menu.css:419]: Bootstrap should be available to all pages already. Please check and confirm why this is needed.

File: build.html (hunk 2 above)
Comment (YaroslavOvdii) [reply build.html:44]: @tonytlwu There is a little problem with the transition in this menu.
As I understand from the [description](https://github.com/Fliplet/fliplet-menu-default#fliplet-default-menu---right) when we click JQ automatically adds an active class. Is there any listener that we could use to change that behavior?
Why do we need this. In order to fix [this issue](https://weboo.atlassian.net/jira/software/projects/OD/boards/9?selectedIssue=OD-356), we need to set the main menu container as hidden (e.g `display: none`) but the [listener](https://github.com/Fliplet/fliplet-menu-default/pull/25/files#diff-82df7b14712db47ddbc76d0b8bc9ff3977cc8e1c2217029413661176afc37f97R67) fires after active class was added and animation is done before we even can remove the hidden class.

Comment (tonytlwu) [reply build.html:44]: @YaroslavOvdii How is this different from the other menus? I see that in the other menus you were able to use `setTimeout` [to achieve the desired outcome](https://github.com/Fliplet/fliplet-menu-circle/pull/24/files). Is it because they are not using `data-fl-toggle-menu`? If that's the case, you can change this menu to not use `data-fl-toggle-menu` and use something like `nav-toggle` like the other menus do.

Comment (YaroslavOvdii) [reply build.html:44]: @tonytlwu Fixed animation

File: js/menu.js [hunk 3]
```diff
@@ -64,16 +64,19 @@ function init() {
     });
   });
 
-  $('[data-fl-toggle-menu]').on('click keydown', function(event) {
+  $('.nav-right.focus-outline').on('click keydown', function(event) {
```
Comment (tonytlwu) [inline js/menu.js:67]: Can we limit this to the menu, e.g.

```suggestion
  $menuElement.find('.nav-right.focus-outline').on('click keydown', function(event) {
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/menu.js (hunk 3 above)
Comment (YaroslavOvdii) [reply js/menu.js:67]: @tonytlwu updated code.

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: App users with the Overlay menu
Verify: Users can access all menu items as expected

Squash and merge: Enhance the accessibility of menu
